modules/network/src/main/java/org/apache/ignite/network/scalecube/ScaleCubeTopologyService.java

```diff
@@ -30,49 +32,62 @@
  * Implementation of {@link TopologyService} based on ScaleCube.
  */
 final class ScaleCubeTopologyService extends AbstractTopologyService {
-    /** Inner representation a ScaleCube cluster. */
-    private Cluster cluster;
+    /** Local member node. */
+    private ClusterNode localMember;
+
+    /** Topology members. */
+    private Map<String, ClusterNode> members = new HashMap<>();
```

Comment:
are you sure it shouldn't be a `ConcurrentHashMap`?

Reply:
True, it should be